### DISCUSSION TITLE
kick off a job run after another job completes without using Assets

### DISCUSSION THREAD

**Discussion by yuhan** [dagster-io/dagster#7160] [Q&A] [answered]
Question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1647374649161559). Reposting it here for posterity and discoverability.

What is the appropriate way to construct a sensor that yields a `job_2` run after any completed `job_1`  run, if i am not using Assets?
is this the right approach?
```
@op
def do_a_thing():
    print("foo")

@job
def do_the_first_thing():
    do_a_thing()

@job
def do_the_second_thing():
    do_a_thing()

@run_status_sensor(
    pipeline_run_status=PipelineRunStatus.SUCCESS, 
    job_selection=[do_the_first_thing], 
)
def thing_sensor(context):
    yield RunRequest(
        job_name=do_the_second_thing
    )
```

**Comment by yuhan** [comment, marked as answer]
Run status sensors currently don’t support yielding RunRequest. So, I'd recommend to write a custom [sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#defining-a-sensor) that listens to the run storage and yields a RunRequest when a `job_1` run finishes.

Something like:
```python
from dagster import RunsFilter, sensor, DagsterRunStatus

@sensor(job=do_the_second_thing)
def thing_sensor(context):
    run_records = context.instance.get_run_records(
        filters=RunsFilter(
            job_name="do_the_first_thing",
            statuses=[DagsterRunStatus.SUCCESS],
            updated_after=..., # can also filter by timestamp to do more efficient fetching
        ),
        order_by="update_timestamp",
        ascending=False,
    )
    for run_record in run_records:
        yield RunRequest(
            run_key=run_record.dagster_run.run_id,  # avoid double firing for the same run
        )
```

**Reply by nvinhphuc** [reply]
This gives error: ```AttributeError: 'RunRecord' object has no attribute 'run_id'```

**Reply by alangenfeld** [reply]
`run_record.pipeline_run.run_id` should work

**Comment by hec10r** [comment]
Hi @yuhan, thanks for the examples. Question: what is the problem with the first approach? Or why the second one is preferred? Thanks!

**Reply by tungp99** [reply]
Same question. Was run_status_sensor designed only for telemetry purpose?

**Reply by yuhan** [reply]
Mainly because run_status_sensor currently doesn't support yielding `RunRequest` so the first approach won't work. I updated the answer to make that clear.
